pkg/armhelpers/azureclient.go

```diff
@@ -211,6 +211,20 @@ func NewAzureClientWithClientCertificate(env azure.Environment, subscriptionID,
 		return nil, err
 	}
 
+	return internalnewAzClientWithCertificate(env, oauthConfig, subscriptionID, clientID, tenantID, certificate, privateKey)
+}
+
+// NewAzureClientWithClientCertificateExternalTenant returns an AzureClient via client_id and jwt certificate assertion against a 3rd party tenant
+func NewAzureClientWithClientCertificateExternalTenant(env azure.Environment, subscriptionID, tenantID, clientID string, certificate *x509.Certificate, privateKey *rsa.PrivateKey) (*AzureClient, error) {
+	oauthConfig, err := adal.NewOAuthConfig(env.ActiveDirectoryEndpoint, tenantID)
+	if err != nil {
+		return nil, err
+	}
+
+	return internalnewAzClientWithCertificate(env, oauthConfig, subscriptionID, clientID, tenantID, certificate, privateKey)
+}
+
+func internalnewAzClientWithCertificate(env azure.Environment, oauthConfig *adal.OAuthConfig, subscriptionID, clientID, tenantID string, certificate *x509.Certificate, privateKey *rsa.PrivateKey) (*AzureClient, error) {
```

Comment:
making this "newAzClientWithCertificate" would be idiomatic and self-explanatory.

Reply:
Let's change it to `newAzureClientWithCertificate` or `internalNewAzureClientWithCertificate` (not sure how important the internal part it, I'll defer to @JackQuincy for that) for consistency with other function names in this file